PMax Assets: Adjust the tip of the disabled "Add image" buttons

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's a follow-up and relates to #1787.

In the last section of https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1439874287, the wording may confuse users:

![2023-02-22 19 24 55](https://user-images.githubusercontent.com/17420811/220606952-74f9a89b-362b-4e35-91bb-013051baee22.png)

The conclusion is to use different tip messages based on the condition:
- If the total number of the selected images is less than the shared maximum number, the tip will include the minimum image requirements.
- Otherwise, the tip only tells the limit of the maximum number.

### Screenshots:

#### 📷 All buttons are disabled

![image](https://user-images.githubusercontent.com/17420811/221817213-b6129bd0-0eb7-4306-b5f9-7eade317a474.png)

#### 📷 Only some buttons are disabled

![image](https://user-images.githubusercontent.com/17420811/221817437-e665ba84-2760-4146-83dc-0bf17975575d.png)

### Detailed test instructions:

💡  Considering the max limit of 20 is not that easy to test, it would be easier by changing the number in the following line to a smaller one:

https://github.com/woocommerce/google-listings-and-ads/blob/3fb9600eb5cd92cd5805378388aa89796d891457/js/src/components/paid-ads/assetSpecs.js#L147

1. Go to step 2 of the campaign creation or editing page.
2. Select 19 or **N** images for the square and portrait fields to make "Add image" buttons partially disabled.
   - **N**: The number of locally adjusted --> `ASSET_MARKETING_IMAGE_SPECS[ sharedMaxSymbol ] - 1`
3. Hover over the disabled button. The tooltip should include the minimum requirements.
4. Select the last image to make all buttons disabled.
5. Hover over the disabled button. The tooltip should not include the minimum requirements.

### Additional details:

Related Figma: fqR0EHi63lWahRcVTKCcba-fi-5434%3A227215

### Changelog entry

eason9487 committed:
Change the tip of the disabled "Add image" buttons to be different based on the condition.

Address
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1439874287

Ref
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1439596853
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1443123179
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1443842423
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1444219161
- https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1447670541

eason9487 committed:
Add tests for the `getMaxNumberTip` function.

eason9487 committed:
Tweak the tooltip style in the `ImagesSelector` component.

puntope commented on js/src/components/paid-ads/assetSpecs.js:
```diff
@@ -435,6 +438,59 @@ const ASSET_TEXT_SPECS = [
+		);
+	}
+
+	// Maximum 20 images can be uploaded, with a minimum of 1 landscape and 1 square image.
```
💅 I this comment deliberately added here?

eason9487 replied:
Thanks for noticing this. That's my temporary note when developing. Removed by https://github.com/woocommerce/google-listings-and-ads/pull/1911/commits/ceeb912b3ba4e7aac10d832e550586c71b039670.

puntope reviewed (approved):
✅  LGTM Thanks for the tweak

I left one comment in regards one of your comments.

eason9487 committed:
Remove a temporary dev note from the `assetSpecs` module.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1911#discussion_r1121799330
